app/src/main/java/io/novafoundation/nova/app/root/navigation/swap/SwapNavigator.kt

```diff
@@ -12,4 +16,16 @@ class SwapNavigator(
     override fun openSwapConfirmation() {
         navigationHolder.navController?.navigate(R.id.action_swapMainSettingsFragment_to_swapConfirmationFragment)
     }
+
+    override fun selectAssetIn(selectedAsset: FullChainAssetId?) {
+        val selectedAssetModel = selectedAsset?.let { AssetPayload(it.chainId, it.assetId) }
```

Comment:
I think we can pass `AssetPayload` directly to remain consistent with other places we pass some payload for navigation